contracts/AddressDiscovery.sol

```diff
@@ -2,14 +2,30 @@
 pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts/access/AccessControl.sol";
-import "./CBDCAccessControl.sol";
 
-contract AddressDiscovery is CBDCAccessControl {
```

Comment:
`AddressDiscovery` can't be `CBDCAccessControl`, because it would also have to expose the other roles that `CBDCAccessControl` exposes (MINTER, BURNER, etc). Those roles are not in the `AddressDiscovery` ABI, so I think Bacen used plain OpenZeppelin `AccessControl` here.